### PR TITLE
kaldi import/export -- adding basic tests for kaldi import/export

### PR DESCRIPTION
**Pull request by jtrmal**
I'm fairly new to pytest -- please let me know if it's merge-worthy or should we re-done in some other way.
I will continue on adding more tests/checks

### CODE REVIEW

**Comment by jtrmal** [comment]
The fails are just flac missing. Not important for now
y.

**Comment by pzelasko** [comment]
Looks good! Do you think we can get away with a smaller number of files, like 2-3 FLACs? It would make the clone/download size much smaller.

**Comment by csukuangfj** [comment]
How about generating the sound files dynamically with sox?

**Comment by jtrmal** [comment]
I removed extra files that I do not have tests for yet (feature-files related mostly) and implemented the sox suggestion (which is a great thinking!). @pzelasko  ready for review/merge if you like

**Comment by jtrmal** [comment]
ugh, not ready yet, sorry!

**Comment by jtrmal** [comment]
I think this is pretty much all the test cases I was able to figure out. let me know.

File: test/test_kaldi_dirs.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
+import pytest
+import os.path
```
Comment (csukuangfj) [inline test/test_kaldi_dirs.py:2]: People usually use `import os`, I think.
After that, you can use `os.path` in the code.

Also, people usually sort the imports within a block alphabetically, i.e.,
```python3
import contextlib
import os
import pytest
```

File: test/test_kaldi_dirs.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+import contextlib
+from pathlib import Path
+
+import lhotse
+import lhotse.utils
+from lhotse.utils import is_module_available
+
+FIXTURE_PATH = Path(os.path.realpath(__file__)).parent / "fixtures"
```
Comment (csukuangfj) [inline test/test_kaldi_dirs.py:10]:
```suggestion
FIXTURE_PATH = Path(__file__).resolve().parent / "fixtures"
```

and you can use
```python
Path('xxx').is_file()  # or Path('xxx').exists()
```
in the following and remove `import os.path`.

File: test/test_kaldi_dirs.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+                text="SOMETHING ELSE",
+                speaker="lbi-1272-135031-B",
+                gender="f",
+            ),
+        ]
+    )
+    return recording, supervision
+
+
+@pytest.mark.xfail(reason="multi fail recordings not supported yet")
```
Comment (csukuangfj) [inline test/test_kaldi_dirs.py:112]:
```suggestion
@pytest.mark.xfail(reason="multi file recordings not supported yet")
```

File: test/test_kaldi_dirs.py [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+def test_multi_file_recording(tmp_path, multi_file_recording):
+    with working_directory(tmp_path):
+        lhotse.kaldi.export_to_kaldi(
+            multi_file_recording[0], multi_file_recording[1], ".", None, False
```
Comment (pzelasko) [inline test/test_kaldi_dirs.py:116]: can you use keyword args here (and other places)? I'm not sure what ".", None, and False correspond to when reading the test

File: test/test_kaldi_dirs.py [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+    with working_directory(fixture_path / lhotse_dir):
+        recording_set = lhotse.RecordingSet.from_jsonl("recordings.jsonl.gz")
+        supervision_set = lhotse.SupervisionSet.from_jsonl("supervisions.jsonl.gz")
+
+    assert out[0] == recording_set
+    assert out[1] == supervision_set
+
+
+@pytest.mark.xfail(reason="Import ordering is somehow not working for now")
+def test_get_duration_failurex(monkeypatch):
```
Comment (pzelasko) [inline test/test_kaldi_dirs.py:157]: will fix imports (for future self: `from augmentation import *` is broken in top-level `__init__.py`)

File: test/test_kaldi_dirs.py [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+
+def test_multi_channel_recording(tmp_path, multi_channel_recording):
+    with working_directory(tmp_path):
+        lhotse.kaldi.export_to_kaldi(
+            multi_channel_recording[0], multi_channel_recording[1], ".", None, False
+        )
+
+
+@contextlib.contextmanager
+def working_directory(path):
```
Comment (pzelasko) [inline test/test_kaldi_dirs.py:128]: nice readability trick

**Review by pzelasko — COMMENTED** [review]
Thanks a lot, looks good to me! Can you address the kwarg comment?

**Comment by jtrmal** [comment]
thanks for the comments, will resolve it tomorrow.
y.

**Comment by jtrmal** [comment]
yeah, will do
y.

**Comment by jtrmal** [comment]
@pzelasko @csukuangfj  I addressed your feedback

**Pull request merged**